src/module.js

```diff
@@ -1195,6 +1201,54 @@ export default class DailyIframe extends EventEmitter {
     return this;
   }
 
+  setShowLocalVideo(bool = true) {
+    methodNotSupportedInReactNative();
+    if (typeof bool !== 'boolean') {
+      console.error('setShowLocalVideo only accepts a boolean value');
+      return this;
+    }
+    if (this._callObjectMode) {
+      console.error('setShowLocalVideo is not available in callObject mode');
+      return this;
+    }
+    this.sendMessageToCallMachine({
+      action: DAILY_METHOD_SET_SHOW_LOCAL_VIDEO,
+      bool,
+    });
+    this._showLocalVideo = bool;
```

Comment:
hm, by tracking this state with an instance variable here (`_showLocalVideo`), we open the door to the possibility that a customer will invoke `setShowLocalVideo()` *before* we've finished joining the call. if that happens, the state tracked by `_showLocalVideo` could get out of sync and report the wrong thing. also, this instance variable might report the wrong thing if the `showLocalVideo` property is passed in when joining the call in the first place, since i don't see where that property is synced with `_showLocalVideo`.

sorry, looks like this is turning into a bit more work :(

i see two paths forward:
* disallow invoking `setShowLocalVideo()` if the call hasn't finished joining (which you can check with `this._meetingState !== DAILY_STATE_JOINED`), and make sure that you assign the `showLocalVideo` join property to `this._showLocalVideo`.
* make the `showLocalVideo()` function async—more like `getInputDevices()` or `getNetworkStats()`—where it fetches the state from the call machine, meaning you don't have to track it as an instance variable here at all.

Reply:
on further reflection, the second approach seems much less ergonomic for UI implementers. much easier to have a synchronous state lookup in react...

Reply:
Ya, that makes sense. To make sure I'm reading your second comment right, you're suggesting this first approach then?

> disallow invoking setShowLocalVideo() if the call hasn't finished joining (which you can check with this._meetingState !== DAILY_STATE_JOINED), and make sure that you assign the showLocalVideo join property to this._showLocalVideo.

Reply:
Pushed this change (the first option you mentioned). I'll update if I misread what you meant